### ISSUE TITLE
Does not work on Fedora 23

### ISSUE THREAD

**Issue by urbenlegend** [geoffreytran/AX88179_178A#3]
I compiled the kernel module just fine and was able to modprobe it. However, it doesn't seem to activate the Ethernet port on my Belkin B2B128. I don't see any logs in journalctl that show an error.

**Comment by geoffreytran** [comment]
Does the Belkin get detected properly?

What does the output of lsusb show?

**Comment by urbenlegend** [comment]
The USB ports on the hub work fine. Its just the Ethernet port that isn't working.

This is what lsusb shows when its pluggedin:
us 001 Device 003: ID 04ca:7036 Lite-On Technology Corp. 
Bus 001 Device 002: ID 8087:8000 Intel Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 006: ID 17ef:304f Lenovo 
Bus 003 Device 005: ID 2109:0812 VIA Labs, Inc. VL812 Hub
Bus 003 Device 004: ID 2109:0812 VIA Labs, Inc. VL812 Hub
Bus 003 Device 003: ID 050d:0128 Belkin Components 
Bus 003 Device 002: ID 2109:0812 VIA Labs, Inc. VL812 Hub
Bus 003 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 004: ID 8087:07dc Intel Corp. 
Bus 002 Device 003: ID 138a:0017 Validity Sensors, Inc. Fingerprint Reader
Bus 002 Device 009: ID 413c:2112 Dell Computer Corp. 
Bus 002 Device 007: ID 046d:c063 Logitech, Inc. DELL Laser Mouse
Bus 002 Device 006: ID 1a40:0101 Terminus Technology Inc. Hub
Bus 002 Device 011: ID 17ef:304e Lenovo 
Bus 002 Device 010: ID 0424:2514 Standard Microsystems Corp. USB 2.0 Hub
Bus 002 Device 008: ID 2109:2812 VIA Labs, Inc. VL812 Hub
Bus 002 Device 005: ID 2109:2812 VIA Labs, Inc. VL812 Hub
Bus 002 Device 002: ID 2109:2812 VIA Labs, Inc. VL812 Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

This is what is shows when it isn't:

Bus 001 Device 003: ID 04ca:7036 Lite-On Technology Corp. 
Bus 001 Device 002: ID 8087:8000 Intel Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 006: ID 17ef:304f Lenovo 
Bus 003 Device 005: ID 2109:0812 VIA Labs, Inc. VL812 Hub
Bus 003 Device 004: ID 2109:0812 VIA Labs, Inc. VL812 Hub
Bus 003 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 004: ID 8087:07dc Intel Corp. 
Bus 002 Device 003: ID 138a:0017 Validity Sensors, Inc. Fingerprint Reader
Bus 002 Device 009: ID 413c:2112 Dell Computer Corp. 
Bus 002 Device 007: ID 046d:c063 Logitech, Inc. DELL Laser Mouse
Bus 002 Device 006: ID 1a40:0101 Terminus Technology Inc. Hub
Bus 002 Device 011: ID 17ef:304e Lenovo 
Bus 002 Device 010: ID 0424:2514 Standard Microsystems Corp. USB 2.0 Hub
Bus 002 Device 008: ID 2109:2812 VIA Labs, Inc. VL812 Hub
Bus 002 Device 005: ID 2109:2812 VIA Labs, Inc. VL812 Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Comment by urbenlegend** [comment]
Whoops, my bad. Seems like my system already provides a ax88179_178a.ko.xz. I am assuming this is the one that doesn't support the Belkin B2B128. How do I make the system take the new ax88179_178a without modifying files provided by my kernel package?

**Comment by urbenlegend** [comment]
Yep I can confirm. If I use the manual modprobe procedure described in the readme, the Belkin works fine. This should be mentioned in the README. If the system already provides ax88179_178a, it will interfere with the one that's installed via "make install".

**Comment by urbenlegend** [comment]
Any chance that this driver will get into the mainline kernel?

**Comment by geoffreytran** [comment]
I'm not sure, I have not attempted to try and submit it. If you know how and would like to submit it, that would be great.

Geoffrey Tran

**Comment by urbenlegend** [comment]
I am not familiar with kernel development myself, but you might want to try asking around on the linux-kernel mailing lists. http://www.tux.org/lkml

**Comment by urbenlegend** [comment]
Actually a better way would be to upstream it to the ASIX guys themselves and have them integrate the code. Then the linux kernel would just pull from them.
